src/nbl/video/COpenGL_Swapchain.h

```diff
@@ -177,7 +184,10 @@ class COpenGL_Swapchain final : public ISwapchain
         {
             egl->call.peglBindAPI(FunctionTableType::EGL_API_TYPE);
 
-            egl->call.peglMakeCurrent(egl->display, surface, surface, thisCtx);
+            EGLBoolean mcres = m_makeCurrentRes = egl->call.peglMakeCurrent(egl->display, surface, surface, thisCtx);
+            assert(mcres == EGL_TRUE);
+
+            m_ctxCreatedCvar.notify_one();
```

Comment:
are you 100% sure that the un-current main & aux context creation + aux context make current is atomic?

I.e. between aux context creation and aux context make current (the duration of time when the main context is not current) nobody can come around and try to make the main thread do any GL work?

Reply:
main thread never does any GL work, but i presume you meant master ctx/device thread. Answering the question.. hm, not from main but other threads could submit some work inbetween unbind and bind device context. Good catch, i'll fix that

Reply:
yeah that was the burning red flag about unbinding the master context for any window of time.

Reply:
@Crisspl so now that I'm vindicated with my warnings about not signalling outside of a mutex, can you check this for fuckups?